### PR TITLE
[Core] Read rerun files from directories

### PR DESCRIPTION
**Pull request by dbfwhitney**
Added a method to try and add functionality to process all rerun folders in a directory (if more than 0 files exist). Still needs work as not sure how to implement walkFileTree method.

### CODE REVIEW

File: cucumber-core/src/main/java/io/cucumber/core/options/CucumberPropertiesParser.java
Comment (mpkorstanje) [inline cucumber-core/src/main/java/io/cucumber/core/options/CucumberPropertiesParser.java:206]: Consider returning a `Set<Path>` instead. The file visitor already provides you with `Path`s and the `parseFeatureWithLinesFile` also consumes paths.

File: cucumber-core/src/main/java/io/cucumber/core/options/CucumberPropertiesParser.java
Comment (mpkorstanje) [inline cucumber-core/src/main/java/io/cucumber/core/options/CucumberPropertiesParser.java:213]: If I'm not mistaken, the `Path file` is a complete path the the rerun file. So you can directly add it to the the `fileList`.

File: cucumber-core/src/main/java/io/cucumber/core/options/CucumberPropertiesParser.java
Comment (mpkorstanje) [reply cucumber-core/src/main/java/io/cucumber/core/options/CucumberPropertiesParser.java:206]: And to make reading this code a bit easier, putting this method after `parseRerunFiles` would be nice.

**Review by mpkorstanje — CHANGES_REQUESTED** [review]
Looks good! Now there are a few more things that need to be done before this can be merged.

1. Write some tests.
2. Clean up the code
3. Update the `CHANGELOG.md`

Once that is done the feature can be released.

## Writing tests

The test for this class would go into the `CucumberPropertiesParserTest`. There already is a test for `should_parse_rerun_file` so you can use that to pattern a few more tests. I reckon the important cases are a directory with no files and a directory with two files.

These tests will make the next part easier.

## Cleaning up the code

I've left some comments on the code itself. They'll make the code a bit better.

## Update the CHANGELOG

In the `CHANGELOG.md` file you can add a `### Fixed` header and add an entry for your change. Have a look at the older entries to see the format used.

**Comment by mpkorstanje** [comment]
Pushed some changes. Still have to update the docs on `@CucumberOptions`, `USAGE.txt`, `Constants.java`, ect.

**Review by mpkorstanje — APPROVED** [review]

**Comment by aslakhellesoy** [comment]
Hi @dbfwhitney,

Thanks for your making your first contribution to Cucumber, and welcome to the Cucumber committers team! You can now push directly to this repo and all other repos under the cucumber organization! 🍾

In return for this generous offer we hope you will:

* ✅ Continue to use branches and pull requests. When someone on the core team approves a pull request (yours or someone else's), you're welcome to merge it yourself.
* 💚 Commit to setting a good example by following and upholding our [code of conduct](https://cucumber.io/conduct) in your interactions with other collaborators and users.
* 💬 Join the [community Slack channel](http://cucumber.io/support#slack) to meet the rest of the team and make yourself at home.
* ℹ️ Don't feel obliged to help, just do what you can if you have the time and the energy.
* 🙋 Ask if you need anything. We're looking for feedback about how to make the project more welcoming, so please tell us!

On behalf of the Cucumber core team,
Aslak Hellesøy
Creator of Cucumber

**Comment by mpkorstanje** [comment]
@dbfwhitney thanks for the first setup.

**Pull request merged**
